### ISSUE TITLE
lost ownership and ACL when file is moved to quarantine

### ISSUE THREAD

**Issue by BlackZork** [Cisco-Talos/clamav#844]
Describe the bug
----------------
I run clamdscan nightly with following parameters:
clamdscan --multiscan --fdpass /samba/some_dir --move /root/quarantine

/root/quarantine and /samba/some_dir are on different filesystems - both ext4, both support extended ACL (getfacl/setfacl).

If there is a false positive, clamdscan moves file to quarantine. I see in logs:

```
Feb 20 00:41:56 universe clamdscan[981646]: /samba/sdelete64.exe: Sanesecurity.Malware.28891.BadCo.UNOFFICIAL FOUND
Feb 20 00:41:56 universe clamdscan[981646]: traverse_rename: Failed to rename: /samba/sdelete64.exe
Feb 20 00:41:56 universe clamdscan[981646]:         to: /root/quarantine/sdelete64.exe
Feb 20 00:41:56 universe clamdscan[981646]: Error:Invalid cross-device link
```

File is moved to quarantine, but is owned by root and all ACL are lost. I have to move file back from quarantine and then restore its ACL from backup. I would rather want clamdscan to preserve all file atributes when moved to quarantine. I would be ideal if files moved to quarantine were placed with full paths too. 

Then if there is a false positive I could just mv file back.

**Comment by Sanesecurity** [comment]
I've fixed the above FP... Sanesecurity.Malware.28891.BadCo.UNOFFICIAL  so if you retest and it's not there you know what happened

**Comment by micahsnyder** [comment]
> I would rather want clamdscan to preserve all file atributes when moved to quarantine.

This makes sense to me.  I'll set this issue to "feature request."  If anyone works on it, pelease make sure the change applies to `clamdscan` as well as `clamscan` and `clamonacc`.  The code is would probably go under the `common` directory and apply to all of them automatically, but just check that it does. 

>  I would be ideal if files moved to quarantine were placed with full paths too.

Do you mean that you would prefer if it went to `/root/quarantine/samba/sdelete64.exe`?  I'm not sure about this change.  Though I can see the need to easily know where a file came from so it can be restored.  Perhaps a log message "to restore, run this command" type of thing.  (Just brainstorming)

**Comment by BlackZork** [comment]
Yes, I would like to at least have an information from which path a file is coming from. This is important when in case of false positive multiple files with the same name are put into quarantine (not sure what will happen now - they are renamed to avoid overwriting?)
